kotlinx-coroutines-core/common/src/sync/Mutex.kt

```diff
@@ -189,12 +191,14 @@ internal open class MutexImpl(locked: Boolean) : SemaphoreImpl(1, if (locked) 1
                 // locked by our owner.
                 if (owner != null) {
```

Comment:
No, in case "somebody else owns the lock", `tryLock()` is eligible to return `false`.

The problem is that 
0. the mutex was locked with the same owner
1. `tryAcquire()` fails
<another thread releases the mutex>
2. `holdsLock(o)` fails, as the mutex is unlocked
<another thread acquires the mutex with owner `o`>
3. `isLocked` returns `true`, and `tryLock(o)` returns `false`. However, `tryLock(o)` should throw an exception.